Sdss 1347 workgroup tagging

ksharp-drupal opened the pull request:
# READY FOR REVIEW

# Summary

- Adds su_person_sunetid field to person type and includes it in the migration
- Implements person tagging from workgroups to new field su_sdss_person_wg_tags

# Review By (Date)
- In time for next testing build.

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Run person importer, verify that sunetids have been populated
3. Configure some workgroup tags under Importers menu
    3.1. Try entering a workgroup more than once to trigger form error
    3..2 Try adding the same tag for two different workgroups with conflicting 'auto-remove' check to trigger form error.
4. Run tagger under importers menu and verify tagging.
5. Check if ultimate cron setting for tagger is correct.
6. Run config import to make sure config_ignore and core.extension settings are correct.
7.  Profit.

joegl reviewed (changes requested):
Initial review with some thoughts and suggestions. Am going to pull down to test soon.

joegl commented:
@ksharp-drupal You may have noticed I made a few changes to this PR. I did **not** touch any of the core functionality code. The changes I made include:

- Cleaned up the README, LICENSE, version numbers, etc.
- Added a few minor fixes.
- Refactored the Controller and Workgroup Tagging Utilities into Drupal services and added dependency injection.
- Renamed a few variables for clarity.
- Added PHPUnit tests.
- Added an "enable" checkbox to the configuration form. This allows us to deploy and enable the module across all sites, but only those that have enabled it in the settings and configured it will actually run it.

I tested the functionality and was able to get the tagging working. However, it appears that the auto-remove functionality is not working. I tried to find where the code uses the 'remove' array keys to actually remove terms, but I didn’t see anything beyond setting the `$updated` variable to `TRUE`. That said, I’m happy to merge and deploy this initial version without that feature. In fact, auto-removal could be unnecessarily destructive at first, since some site owners may have manually tagged people and wouldn’t want to lose that work. We should get input from the rest of the SDSS team on this.

Let me know your thoughts—thanks!

ksharp-drupal commented:
Thanks, Joe! That all sounds good. I’ll try to run some quick tests before our sprint meeting today. -Kassie
-------------------------------------
Kassie Sharp (she/her/hers)
Senior Web Developer
Stanford Doerr School of Sustainability
(650) 721-6406
***@***.***



On Jun 4, 2025, at 10:48 AM, Joe Gilliland-Lloyd ***@***.***> wrote:

[https://avatars.githubusercontent.com/u/6943710?s=20&v=4]joegl left a comment (SU-SWS/ace-sdssgryphon#581)<https://github.com/SU-SWS/ace-sdssgryphon/pull/581#issuecomment-2940880065>

@ksharp-drupal<https://github.com/ksharp-drupal> You may have noticed I made a few changes to this PR. I did not touch any of the core functionality code. The changes I made include:

  *   Cleaned up the README, LICENSE, version numbers, etc.
  *   Added a few minor fixes.
  *   Refactored the Controller and Workgroup Tagging Utilities into Drupal services and added dependency injection.
  *   Renamed a few variables for clarity.
  *   Added PHPUnit tests.
  *   Added an "enable" checkbox to the configuration form. This allows us to deploy and enable the module across all sites, but only those that have enabled it in the settings and configured it will actually run it.

I tested the functionality and was able to get the tagging working. However, it appears that the auto-remove functionality is not working. I tried to find where the code uses the 'remove' array keys to actually remove terms, but I didn’t see anything beyond setting the $updated variable to TRUE. That said, I’m happy to merge and deploy this initial version without that feature. In fact, auto-removal could be unnecessarily destructive at first, since some site owners may have manually tagged people and wouldn’t want to lose that work. We should get input from the rest of the SDSS team on this.

Let me know your thoughts—thanks!

—
Reply to this email directly, view it on GitHub<https://github.com/SU-SWS/ace-sdssgryphon/pull/581#issuecomment-2940880065>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAGDBISDMIAPAFAKRRB7YAT3B4WNHAVCNFSM6AAAAAB4PK76FKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNBQHA4DAMBWGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
